pkg/provider/provider.go

```diff
@@ -558,14 +555,14 @@ var (
 	configureClientError error //nolint:errname
 )
 
-func ConfigureProvider(s *schema.ResourceData) (interface{}, error) {
+func ConfigureProvider(ctx context.Context, s *schema.ResourceData) (any, diag.Diagnostics) {
 	// hacky way to speed up our acceptance tests
-	if os.Getenv("TF_ACC") != "" && os.Getenv("SF_TF_ACC_TEST_CONFIGURE_CLIENT_ONCE") == "true" {
+	if os.Getenv(resource.EnvTfAcc) != "" && os.Getenv(string(testenvs.ConfigureClientOnce)) == "true" {
```

Comment:
we should not have the dependecy to testenvs from this production code (that's the reason for which it was kept like this).

Also the dependency to github.com/hashicorp/terraform-plugin-testing/helper/resource" should be removed.

Reply:
Done.